### ISSUE TITLE
Template receiver: Pool and Proxy JN

### ISSUE THREAD

**Issue by 4ss0** [stratum-mining/stratum#417]
handle_setup_connection_error and  handle_channel_endpoint_changed are two functions to be implemented. 
They are present in roles\v2\pool\src\lib\mining_pool\setup_connection.rs and roles\v2\mining-proxy\src\lib\template_receiver\setup_connection.rs

**Comment by lorbax** [comment]
I think this needs some attention.
@Fi3  @pavlenex

**Comment by pavlenex** [comment]
@lorbax what should we do about this one, the issue @4ss0 logged doesn't bring enough context for me to understand what needs to be done?

**Comment by lorbax** [comment]
You are right, there is little to no context here. I think that "setup_connection_error and handle_channel_endpoint is unimplemented" refers to the todo!() at line 85 and 92 here
[https://github.com/stratum-mining/stratum/blob/dev/roles/jd-client/src/job_declarator/setup_connection.rs)](https://github.com/stratum-mining/stratum/blob/dev/roles/jd-client/src/job_declarator/setup_connection.rs)
I am still trying to figure out if these messages shall be implemented with priority or not

**Comment by Fi3** [comment]
No role at the moment support endpoint changed so maybe we will implement them all together. For the setup error we could either:
1. retry with exponential delay capped at n
2. try the other upstream if any otherwise switch to solo
3. just terminate the JDC with an error

**Comment by GitGab19** [comment]
Closing this as stale
